src/app.js

```diff
@@ -1,12 +1,88 @@
+import axios from "axios";
+import distance from "euclidean-distance";
+
+// Get the API token
+async function getToken() {
+  let token;
+  try {
+    const response = await axios.post(
+      "https://blue-bottle-api-test.herokuapp.com/v1/tokens",
+      {
+        headers: {
+          Accept: "application/json",
+          "Content-Type": "application/json",
+        },
+      }
+    );
+    token = response.data.token;
+  } catch (e) {
+    console.error(`Exception occurred while fetching the token: ${e}`);
+  }
+  return token;
+}
+
+// Get all shop data from the API
+async function getShopData() {
+  const token = await getToken();
+
+  try {
+    const response = await axios.get(
+      `https://blue-bottle-api-test.herokuapp.com/v1/coffee_shops?token=${token}`,
+      {
+        headers: {
+          Accept: "application/json",
+        },
+      }
+    );
+    return response.data;
+  } catch (e) {
+    if (e.response.status == 401 || 406 || 503 || 504) {
+      console.error(`Exception occurred while fetching shop list : ${e}`);
+      return [];
+    }
+  }
+}
+
 /**
  * @param {Object} position
  * @param {Number} position.x
  * @param {Number} position.y
- * 
+ *
  * @returns {Array<position>}
  */
-export function getNearestShops(position) {
-  // code
-  
-  return [];
+
+// Calculate distance of each shop from the given coordinate
+export default async function getNearestShops(position) {
+  const shopData = await getShopData();
```

Comment:
Here you chose to use await but (here)[https://github.com/Agilefreaks/js-interview-starting-point/pull/2/files#diff-bfe9874d239014961b1ae4e89875a6155667db834a410aaaa2ebe3cf89820556R12] you are using `then` syntax, can you please detail and how you made this decision?

Reply:
As you know, top-level await is not supported in older Nodejs versions. (I checked it now, it is supported with Node >14.8). So, to make it more compatible with older Nodejs versions, I used `then` syntax. I could have used some workarounds (compatible with older Nodejs versions) that I encountered on the web but I thought `then` syntax could be the best and easiest way to handle this in the index.js file.